File: Bitwise Operation/191. Number of 1 Bits/main.go
package main

import (
	"fmt"
)

func hammingWeight(num uint32) int {
	count := 0
	for num > 0 {
		num = num & (num - 1)
		count++
	}
	return count
}

func hammingWeight1(num uint32) int {
	count := 0
	mask := uint32(1)
	for i := 0; i < 32; i++ {
		if (num & mask) != 0 {
			count++
		}
		mask = mask << 1
	}
	return count
}

func main() {
	num := uint32(0b11111111111111111111111111111101)
	fmt.Println("Output: ", hammingWeight1(num))
}

File: Bitwise Operation/191. Number of 1 Bits/main.c
#include <stdio.h>
#include <stdint.h>

int hammingWeight(uint32_t n) {
    int count = 0;
	while(n > 0) {
		n = n & (n - 1);
		count++;
	}
	return count;
}

int main() {
    uint32_t n = 0b11111111111111111111111111111101;
    printf("Output: %d.", hammingWeight(n));
    return 0;
}